### PR TITLE
Polkit rules to fix #1201 and improve network permissions avoiding the nm workaround

### LINKED ISSUE #1201: [BUG] Cannot shutdown or reboot host
### What happened?

When trying to shutdown or reboot the host system for the GUI, it shows the following error in the log and just ignores the command:

```
Dec 12 14:08:53 klipperscreen systemctl[1062]: Call to PowerOff failed: Interactive authentication required.
```

I'm using RPI OS 12 (Bookworm) with 64bit rpi3b:

```console
$ uname -a
Linux klipperscreen 6.1.0-rpi7-rpi-v8 #1 SMP PREEMPT Debian 1:6.1.63-1+rpt1 (2023-11-24) aarch64 GNU/Linux
```

and `/etc/X11/Xwrapper.config` contains:

```
# Xwrapper.config (Debian X Window System server wrapper configuration file)
#
# This file was generated by the post-installation script of the
# xserver-xorg-legacy package using values from the debconf database.
#
# See the Xwrapper.config(5) manual page for more information.
#
# This file is automatically updated on upgrades of the xserver-xorg-legacy
# package *only* if it has not been modified since the last upgrade of that
# package.
#
# If you have edited this file but would like it to be automatically updated
# again, run the following command as root:
#   dpkg-reconfigure xserver-xorg-legacy
needs_root_rights=yes
allowed_users=anybody
```

Also note that this KlipperScreen runs on a separate RPI, and Moonraker/Klipper run on another RPI.

### What did you expect to happen instead?

No error, and host shutdown.

### How to reproduce this bug?

Go to system -> shutdown -> host system

### Additional information:

```console
thijs@klipperscreen:~ $ POLKIT_VERSION="$( pkaction --version | grep -Po "(\d+\.?\d*)" )"
thijs@klipperscreen:~ $ echo $POLKIT_VERSION
122
```

### Log output

```shell
● KlipperScreen.service - KlipperScreen
     Loaded: loaded (/etc/systemd/system/KlipperScreen.service; enabled; preset: enabled)
     Active: active (running) since Tue 2023-12-12 01:20:15 CET; 12h ago
   Main PID: 497 (KlipperScreen-s)
      Tasks: 7 (limit: 765)
        CPU: 2min 42.267s
     CGroup: /system.slice/KlipperScreen.service
             ├─497 /bin/bash /home/thijs/KlipperScreen/scripts/KlipperScreen-start.sh
             ├─512 /usr/bin/xinit /home/thijs/.KlipperScreen-env/bin/python /home/thijs/KlipperScreen/screen.py
             ├─549 /usr/lib/xorg/Xorg :0
             └─747 /home/thijs/.KlipperScreen-env/bin/python /home/thijs/KlipperScreen/screen.py

Dec 12 14:08:21 klipperscreen KlipperScreen-start.sh[747]: 2023-12-12 14:08:21,668 [system.py:_already_updated()] - klipper v0.12.0-31
Dec 12 14:08:21 klipperscreen KlipperScreen-start.sh[747]: 2023-12-12 14:08:21,672 [system.py:_already_updated()] - led_effect v0.0.10-1
Dec 12 14:08:21 klipperscreen KlipperScreen-start.sh[747]: 2023-12-12 14:08:21,675 [system.py:_already_updated()] - moonraker v0.8.0-206
Dec 12 14:08:24 klipperscreen KlipperScreen-start.sh[747]: 2023-12-12 14:08:24,268 [KlippyGtk.py:Dialog()] - Showing dialog Shutdown (width=83>
Dec 12 14:08:31 klipperscreen KlipperScreen-start.sh[747]: 2023-12-12 14:08:31,205 [KlippyGtk.py:remove_dialog()] - Removing Dialog
Dec 12 14:08:31 klipperscreen systemctl[1041]: Call to PowerOff failed: Interactive authentication required.
Dec 12 14:08:51 klipperscreen KlipperScreen-start.sh[747]: 2023-12-12 14:08:51,340 [KlippyGtk.py:Dialog()] - Showing dialog Shutdown (width=83>
Dec 12 14:08:53 klipperscreen KlipperScreen-start.sh[747]: 2023-12-12 14:08:53,667 [KlippyGtk.py:remove_dialog()] - Removing Dialog
Dec 12 14:08:53 klipperscreen systemctl[1062]: Call to PowerOff failed: Interactive authentication required.
Dec 12 14:13:53 klipperscreen KlipperScreen-start.sh[747]: 2023-12-12 14:13:53,924 [screen.py:show_screensaver()] - Showing Screensaver
```
```

### LINKED ISSUE #1209: [FR] Make extrusion modifier configurable
### Requested feature

Make extrusion modifiers on the fine tuning screen configurable like Z babystep values are.

### Is your feature request related to a problem?

We are using KlipperScreen on a custom design pellet printer and with some materials 2 % flow change does nothing, so for bigger changes we need to go to the web UI and modify the flow there.

### Describe alternatives you have considered

Changing value via web interface

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by alfrix**

### CODE REVIEW

**Comment by thijstriemstra** [comment]
`#1209` doesn't exist..

**Review by thijstriemstra — APPROVED** [review]
I undid the changes described on https://winaero.com/how-to-enable-shutdown-and-reboot-for-a-normal-user-in-debian-jessie/ and added the rule file from this PR and everything seems to work fine now, awesome!

File: scripts/KlipperScreen-install.sh [hunk 1]
```diff
@@ -152,9 +152,82 @@ install_systemd_service()
...
+        RULE_FILE="${POLKIT_DIR}/KlipperScreen.rules"
+    else
+        echo "PolicyKit rules folder not detected"
+        exit 1
+    fi
+    echo_text "Installing PolicyKit Rules to ${RULE_FILE}..."
+
+    KS_GID=$( getent group klipperscreen | awk -F: '{printf "%d", $3}' )
+    sudo /bin/sh -c "cat > ${RULE_FILE}" << EOF
+// Allow KlipperScreen User to reboot and shutdown the system
```
Comment (thijstriemstra) [inline scripts/KlipperScreen-install.sh:189]: perhaps the ability to scan with wifi should be mentioned here as well

**Comment by alfrix** [comment]
> I undid the changes described on https://winaero.com/how-to-enable-shutdown-and-reboot-for-a-normal-user-in-debian-jessie/ and added the rule file from this PR and everything seems to work fine now, awesome!

Thank you very much for testing :+1:

**Pull request merged**